### DISCUSSION TITLE
How to read HTTP request header when using Funqy?

### DISCUSSION THREAD

**Discussion by cuichenli** [quarkusio/quarkus#34427] [Q&A]
may be mistaken, but from what I understand, it doesn't seem possible to retrieve the HTTP request headers when using the funqy-knative or funqy-http extension. Is there a particular reason why this information cannot be accessed? Thank you!

[a comment by an automated account is not shown]

**Comment by patriot1burke** [comment]
Correct,  Funqy is just vanilla functions.  You're not supposed to be aware of the protocol

**Comment by cuichenli** [comment]
sorry but what about `funqy-http`? i think given then name, it should be based on the `http` protocol and we are supposed to have certain ways to get the headers? thanks!

**Comment by patriot1burke** [comment]
Funqy is a cross-platform function API:  AWS Lambda, Azure Functions, Google Cloud Functions, Cloud Events, HTTP.  Not all of these use the HTTP protocol.  If you want access to http like things, use JAX-RS.

**Comment by cuichenli** [comment]
I understand that Funqy is a cross-platform function API, and not all of its implementations use the HTTP protocol. However, I'm curious about the `funqy-http` implementation. Given its name, it seems reasonable to assume that it is based on HTTP. My question is: if `funqy-http` is using the HTTP protocol, why can't users access the headers? If this feature is missing, would a pull request to add it be welcome? Thanks!

**Comment by patriot1burke** [comment]
The feature is not "missing".  A pull request would not be welcome.  Sorry.  Again, if you want HTTP things, use JAX-RS or some other HTTP library.
